bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetEnergyHandler.java

```diff
@@ -210,4 +253,39 @@ private void updateActivePower(BaseOpenMessage msg) {
             updateState(CHANNEL_POWER, UnDefType.UNDEF);
         }
     }
+
+    /**
+     * Updates current day totalizer
+     *
+     * @param msg the EnergyManagement message received
+     * @throws FrameException
+     */
+    private void updateCurrentDayTotalizer(BaseOpenMessage msg) {
+        Double currentDayEnergy;
+        try {
+            currentDayEnergy = Double.parseDouble(msg.getDimValues()[0]) / 1000d;
+            updateState(CHANNEL_ENERGY_TOTALIZER_DAY, new QuantityType<Energy>(currentDayEnergy, Units.KILOWATT_HOUR));
+        } catch (FrameException e) {
+            logger.warn("FrameException on frame {}: {}", msg, e.getMessage());
+            updateState(CHANNEL_ENERGY_TOTALIZER_DAY, UnDefType.UNDEF);
+        }
+    }
+
+    /**
+     * Updates current month totalizer
+     *
+     * @param msg the EnergyManagement message received
+     * @throws FrameException
+     */
+    private void updateCurrentMonthTotalizer(BaseOpenMessage msg) {
+        Double currentMonthEnergy;
+        try {
+            currentMonthEnergy = Double.parseDouble(msg.getDimValues()[0]) / 1000d;
+            updateState(CHANNEL_ENERGY_TOTALIZER_MONTH,
+                    new QuantityType<Energy>(currentMonthEnergy, Units.KILOWATT_HOUR));
+        } catch (FrameException e) {
```

Comment:
I guess you should also catch NumberFormatException

Reply:
Good point, coverting the String to Double could potentially lead to a  NumberFormatException so I will catch it also. I would modify _updateActivePower()_ as well for similarity.